lib/src/connector/connection.cc

```diff
@@ -315,7 +315,7 @@ void Connection::connectAndWait()
     lth_util::Timer timer {};
     while (connection_state_.load() != ConnectionState::open
            && timer.elapsed_milliseconds() < client_metadata_.ws_connection_timeout_ms) {
-        doSleep();
+        doSleep(2);
```

Comment:
Do we always want to check this frequently? Is there any way we could use an actual wait and make changing connection_state_ an event?